Remove const array and scalar variables from ecma_utf8_string_to_number

akosthekiss opened the pull request:
There is no need for putting `LIT_CHAR_*` constants in const
scalars or arrays, they can be used directly. (There is especially
no need for arrays of 10 elements if only 2 of them are used.)

JerryScript-DCO-1.0-Signed-off-by: Akos Kiss akiss@inf.u-szeged.hu

zherczeg commented:
The patch is good, but do we need to use type casting? Does the compiler complains?

LaszloLango commented:
LGTM, but I agree with @zherczeg. Remove the typecasts if they are not needed.

akosthekiss commented:
@zherczeg @LaszloLango I've removed the casts and rebased the patch to current master.

You were right, the casts were not needed -- which was a surprise to me. I'll apparently have to reiterate some parts of my C studies...

LaszloLango commented:
@akosthekiss, thanks for the update. LGTM

zherczeg commented:
This is the convert everything to "int" rule which is the most annoying rule of C

    short a = 6; a += 2; // Compiler warning. Only the ++ is not a warning...

I would not say it is illogical but definitely not handy.

LGTM
